x-pack/plugins/endpoint/public/applications/endpoint/store/index.ts

```diff
@@ -0,0 +1,21 @@
+/*
```

Comment:
@paul-tavares @kevinlog 
Continuing the discussion on `store` module structure here, so we can 'reply'.

I'm adding some details about the structure I have in my PR. 
![image](https://user-images.githubusercontent.com/35559/71830503-df5d6180-3074-11ea-9273-14831250937d.png)

I'll go over a few files to explain my thought process:

[store/camera/selectors.ts](https://github.com/elastic/kibana/pull/53619/files#diff-14029f347f7ba95a420602c5387bcd6d)
This module defines the camera selectors, they take camera state. Each concern will have its own reducer, state shape, and selectors. 
![image](https://user-images.githubusercontent.com/35559/71831128-49c2d180-3076-11ea-83c1-6cfb77fd9e8f.png)

[store/camera/reducer.ts](https://github.com/elastic/kibana/pull/53619/files#diff-53dea9508f5d9c7f283f6cf4aab24f53)
This defines a reducer for the 'camera' concern. It takes camera state, and any Resolver actions.
![image](https://user-images.githubusercontent.com/35559/71831208-6bbc5400-3076-11ea-8cc8-b408540aa9ea.png)

[store/camera/zooming.test.ts](https://github.com/elastic/kibana/pull/53619/files#diff-5d9de62597219aed4af527874e99cf47)
Unit tests for the 'zooming' behavior of the camera. These tests create a redux store with the camera reducer, dispatch Resolver actions, and assert the values of camera selectors.

This lets me test the business logic in isolation from other concerns (and the DOM, and React.)
![image](https://user-images.githubusercontent.com/35559/71831304-9efee300-3076-11ea-9128-f8fda3408918.png)



[store/reducer.ts](https://github.com/elastic/kibana/pull/53619/files#diff-118fa940615495cf306a955b1341de5b)
This is the top level reducer. It is what is passed in during store creation. Right now its just got 'camera' but once we have interaction, graphing, etc, there will be other reducers there.
![image](https://user-images.githubusercontent.com/35559/71830710-4e3aba80-3075-11ea-95db-a8c999c85d88.png)

[store/selectors.ts](https://github.com/elastic/kibana/pull/53619/files#diff-7dd777f1d03cd310741b16bdfb94291f)
This provides the top level selectors, ones used by React code. Individual concerns (i.e. camera, so far) define selectors that receive *camera* state as their argument. This module reexposes some of those selectors. However selectors used from React will be passing in global state (the internal structure of the reducer, and the state shape of the camera concern are encapsulated.) So this module first selects the 'camera' state from global state, then calls the underlying 'camera' selector. This uses a helper function 'composeSelectors'.

Allowing the camera concern (or any other concern) to define selectors that accept their own state is cool. This allows the camera reducer (which also takes camera state) to call the selectors without having knowledge of global state. Also, the camera selectors can call each other, since they all take camera state. Finally, the location of camera state in the global reducer is hidden from the camera selectors implementation. We could move it around, or even have multiple cameras and things would work just fine.
![image](https://user-images.githubusercontent.com/35559/71831056-18e29c80-3076-11ea-9aaf-065446a3442e.png)


[store/index.ts](https://github.com/elastic/kibana/pull/53619/files#diff-6c641a80a568506501d90d4b9d99340a)
This creates the redux store, sets up redux dev tools, and will eventually initialize middleware. When middleware is initialized, it will be passed a reference to a function that can make HTTP requests to the backend. The middleware functions will respond to actions and sometimes make HTTP requests.
![image](https://user-images.githubusercontent.com/35559/71830562-fd2ac680-3074-11ea-812e-facc39bf47c6.png)

Reply:
@oatkiller - thanks for the comments/feedback.
I had looked at your code in Resolver and it was one of the reasons I had proposed a file/code structure that is more "concern" focused. I think you are advocating for that file/code structure given the benefits described above, correct?

Does anyone else have comments?

⬆️: @kevinlog , @peluja1012 , @kqualters-elastic , @bkimmel , @dplumlee , @parkiino 

Let's make a decision here so we can move forward in establishing the pattern and enable others to start interacting with APIs :)

Reply:
so this means doing option 1?

Reply:
@parkiino - yes 😃 - going with option 1. this first commit has the `endpoint_list` concern which contains some samples as to how one can be defined.